### ISSUE TITLE
Pin mac address to container

### ISSUE THREAD

**Issue by dreh23** [hashicorp/nomad#2650]
We need to run docker containers with a pinned mac address. I would like to see a key in the docker driver configuration.

This could look like this:
```
config {
  mac_address = "02:68:b3:29:da:98"
}
```
This obviously triggers a constraint on group count value.

To pass a list of mac addresses could be useful as well, then nomad has to keep track which mac address is already in use. 
```
mac_addresses = [ "02:68:b3:29:da:98","02:68:b3:29:da:99"]
```

**Comment by schmichael** [comment]
Adding `config` block support for `mac_address` is fairly simple and only requires updating a single method in `client/driver/docker.go`

Having the scheduler validate and track MAC address usage is quite a bit more work and not something I'm sure we want to do (at least in the near-term).

Is the config still useful even if the scheduler lacks mac address validation? If so we might do that first and put the scheduler aspect in a new issue.

**Comment by dadgar** [comment]
I think we will always make the tracking the users responsibility. Right now that would be multiple task groups with unique Macs and in the future it could be maintaining an array of MACs and using HIL (templating) to pick the right one given the count.

**Comment by dreh23** [comment]
Docker leaves the responsibility to the user as well, it's possible to spawn several containers with the same Mac.

**Comment by schmichael** [comment]
Fixed and will be released in 0.6.
